test/GithubApi.Repositories.test.js

```diff
@@ -0,0 +1,54 @@
+require('dotenv').config();
+
+const { StatusCodes } = require('http-status-codes');
+const { expect } = require('chai');
+const axios = require('axios');
+const chai = require('chai');
+const md5 = require('md5');
+
+chai.use(require('chai-subset'));
+
+const urlBase = 'https://api.github.com';
+const githubUserName = 'aperdomob';
+const repoName = 'jasmine-json-report';
+
+describe('Consume GET Services from GITHUB API', () => {
+  it('Get User Data', async () => {
+    const response = await axios.get(`${urlBase}/users/${githubUserName}`);
+    expect(response.status).to.equal(StatusCodes.OK);
+    expect(response.data).to.have.property('name');
+    expect(response.data).to.have.property('company');
+    expect(response.data).to.have.property('location');
+    expect(response.data.name).to.equal('Alejandro Perdomo');
+    expect(response.data.company).to.equal('Perficient Latam');
+    expect(response.data.location).to.equal('Colombia');
+  });
+
+  it('Get Repos List and validate data', async () => {
+    const response = await axios.get(`${urlBase}/users/${githubUserName}/repos`);
```

Comment:
In the response of the previous test, you can get this URL:
![image](https://user-images.githubusercontent.com/10209180/176709019-ef048b00-0d63-4bec-a195-8483ff5ac1ee.png)

You have two options:

1. Define a global variable, that is assigned in the previous test, as the value change after is declared, you should declare it as `let` before the `describe`. This approach has the inconvenience that the tests will depend on one on another, but it is ok for this exercise.
2. The other option is to use the [hooks](https://mochajs.org/#hooks), to initialize the required data in tests (`it`). And nest the describes to reuse the data. One example of the structure of the tests with this approach is this:

```
describe('Consume GET Services from GITHUB API', () => {
  let userResponse;
  
  before(async () => {
    userResponse = await axios.get('${urlBase}/users/${githubUserName}');
  });
  
  it('Get User Data', () => {
    expect(userResponse.status).to.equal(StatusCodes.OK);
    expect(userResponse.data.name).to.equal('Alejandro Perdomo');
    expect(userResponse.data.company).to.equal('Perficient Latam');
    expect(userResponse.data.location).to.equal('Colombia');
  });
  
  describe('Using hypermedia', () => {
    let jasmineRepo;
    before(async () => {
      const repositoriesResponse = await axios.get(`${userResponse.repos_url}`);
      jasmineRepo = repositoriesResponse.data.find(
        (repo) => repo.name === repoName);
    });
    
    it('To Get Repos List and validate data', () => {
      expect(jasmineRepo.name).to.equal('jasmine-json-report');
      expect(jasmineRepo.private).to.equal(false);
      expect(jasmineRepo.description).not.equal(null);
    });
    
    it('Download Repo in a Zip File', async () => {
      // code to verify the zip download
    });
    
    describe('and to verify content', () => {
      // code to verify the readme and to download and verify the md5
    });
  });
});
```